pkg/maps/lbmap/maglev_outer_map.go

```diff
@@ -104,7 +104,7 @@ func MaglevOuterMapTableSize(mapName string) (bool, uint32) {
 	}
 	defer innerMap.Close()
 
-	return true, innerMap.ValueSize() / uint32(unsafe.Sizeof(uint16(0)))
+	return true, innerMap.ValueSize() / uint32(unsafe.Sizeof(uint32(0)))
```

Comment:
hmm, good point, I think this will at least change the map size which seems to trigger a map regeneration? @brb

Reply:
This will trigger the removal of the Maglev maps upon `lbmap.InitMaglevMaps()` invocation. The maps will be removed before the datapath has been regenerated, so for some time the service lookups for new connections will fail.

We could address this by temporarily renaming the old Maglev maps, and then removing it after the datapath reloading. The renaming needs to be conditional, and it would depend on v1->v2 is detected AND oldSize x 2 == newSize.

Reply:
Ah, ignore what I wrote above. The map will be removed once we have removed all references to it (i.e., after reloading the datapath progs). So, all should be fine.

Reply:
A bit confused, so this maglev map is not pinned, aka it's not suppose to survive across restarts?

Reply:
The outer map is pinned, and the inner maps are referenced by the outer map, so they are not removed.

I've just tested the upgrade with Maglev enabled and `time.Sleep()` after the outer map has been removed - it works (the old maps are still there). However, after the upgrade, some requests gets dropped. This is because `LB{4,6}_MAGLEV_MAP_INNER` still has u16 instead of u32 as the val. This needs to be fixed in this PR.

Reply:
Done. Although I still don't think I understand why we do not need to copy the maps (since value type change, wouldn't the bytes be offset to wrong position if we still access old map?)...is there a documentation describing the inner/outer map and how they work?

Reply:
The new outer and inner maps will be created by `pkg/service` upon the startup, so no need to copy. We copy the backend maps though, and this is because we need to restore the backend ID allocator state.

Reply:
I see, so agent would be able to "recover" maglev hashing "state" without refering to the old map?

Reply:
Yes.

Reply:
Thanks, makes sense then, closing this thread since no more action to take (already updated inner map key type).